packages/core/src/database/update-where.ts

```diff
@@ -38,10 +38,25 @@ export const buildUpdateWhere: BuildUpdateWhere = <
   const isKeyOfSchema = isKeyOf(schema);
   const connectKeyValueWithEqualSign = (data: Partial<Schema>) =>
     Object.entries(data)
-      .map(
-        ([key, value]) =>
-          isKeyOfSchema(key) && sql`${fields[key]}=${convertToPrimitiveOrSql(key, value)}`
-      )
+      .map(([key, value]) => {
+        if (!isKeyOfSchema(key)) {
+          return;
+        }
+
+        if (value && typeof value === 'object' && !Array.isArray(value)) {
+          /**
+           * Jsonb || operator is used to shallow merge two jsonb types of data
+           * all jsonb data field must be non-nullable
+           * https://www.postgresql.org/docs/current/functions-json.html
+           */
+          return sql`
+            ${fields[key]}=
+              coalesce(${fields[key]},'{}'::jsonb)|| ${convertToPrimitiveOrSql(key, value)}
```

Comment:
do you mind adding a test case (as an example) on that?
there is `update-where.test.ts`

Reply:
let me think about it. As the current test case in update-where.test.ts does not contain actual data manipulation, only with the mocked response, which technically does not test the SQL itself.

Reply:
@wangsijie added